### PR TITLE
Add context about this project in README

### PR DESCRIPTION
**Pull request by JohanLorenzo**
Modeled after https://github.com/mozilla-releng/mozilla-version/pull/77

`maven-lambda` is a an essential piece of the mobile automation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,3 +32,20 @@ git clone https://github.com/mozilla-releng/maven-lambda
...
+
+The process is quite manual. See [bug 1589065 comment 27](https://bugzilla.mozilla.org/show_bug.cgi?id=1589065#c27) for instance. The idea is:
+
+ 1. let Taskcluster build a push to the master branch
+ 2. Get the direct link to `function.zip`
+ 3. Needinfo CloudOps on Bugzilla with that direct link.
+
+## Debugging
+
+Contact the CloudOps team to get access to the logs. They may refer you to tools like [awslogs](https://github.com/jorgebastida/awslogs).
```
Comment (MihaiTabara) [inline README.md:51]: Note to self: This reminds me it's just the two of us with access there. I should escalate this to CloudOps at some point.

**Review by MihaiTabara — APPROVED** [review]
Thanks for dropping this.

File: README.md [hunk 2]
```diff
@@ -4,6 +4,25 @@
 
 Generate maven metadata on an S3 bucket thanks to an AWS Lambda function.
 
+## Why this project?
+
+The [mozilla-mobile](https://github.com/mozilla-mobile/) teams need a maven repository to host Mozilla's Android libraries, like Geckoview or Android-Components. A maven repository is a regular HTTP server that delivers binaries and metadata files. Nothing is custom in terms of protocol, a maven/gradle client just expects files to [follow a given tree structure](https://blog.packagecloud.io/eng/2017/03/09/how-does-a-maven-repository-work/).
+
+Mozilla could have used an existing product like [Sonatype's Nexus](https://www.sonatype.com/nexus/). Although, we want to ensure the security of these binaries in case an attacker compromises the server. There are 2 main ways to solve this:
+
+ a. sign the binaries to ensure they were modified at rest.
```
Comment (escapewindow) [inline README.md:13]: * not modified

**Review by escapewindow — APPROVED** [review]
Thank you!

File: README.md (hunk 2 above)
Comment (JohanLorenzo) [reply README.md:13]: Oops, thanks! 

```suggestion
 a. sign the binaries to ensure they were not modified at rest.
```

**Pull request merged**